Firestore/core/test/unit/api/aggregate_query_test.cc

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// TODO(b/280805906) Remove these tests for the count specific API after the c++
+// SDK migrates to the new Aggregate API
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "Firestore/core/src/api/aggregate_query.h"
+#include "Firestore/core/src/api/query_core.h"
+#include "Firestore/core/src/util/status.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+
+using model::AggregateAlias;
+using ::testing::Invoke;
+using util::Status;
+
+namespace api {
+
+class MockAggregateQuery : public AggregateQuery {
+ public:
+  MockAggregateQuery(Query query, std::vector<AggregateField>&& aggregates)
```

Comment:
Since the `MockAggregateQuery` is merely delegating to the constructor of the superclass, you can just inherit the superclass' constructors by specifying

```
using AggregateQuery::AggregateQuery;
```

See https://abseil.io/tips/74.